<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Virtual Inference of Protein-activity by Enriched Regulon analysis (preparing input)</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Proteomics and Phosphoproteomics Pipeline</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/zhenzuo2/PPP">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Virtual Inference of Protein-activity by
Enriched Regulon analysis (preparing input)</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-05-24
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>PPP/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240521code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240521)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240521code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240521)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomzhenzuo2PPPtreefe48b9faedf94f04f17e3fef08e1689b4349bdf3targetblankfe48b9fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/zhenzuo2/PPP/tree/fe48b9faedf94f04f17e3fef08e1689b4349bdf3" target="_blank">fe48b9f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomzhenzuo2PPPtreefe48b9faedf94f04f17e3fef08e1689b4349bdf3targetblankfe48b9fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/zhenzuo2/PPP/tree/fe48b9faedf94f04f17e3fef08e1689b4349bdf3" target="_blank">fe48b9f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    data/.DS_Store
    Ignored:    data/viper/.DS_Store
    Ignored:    output/.DS_Store
    Ignored:    output/viper/.DS_Store
    Ignored:    output/viper/exprs/.DS_Store
    Ignored:    output/viper/regulon/.DS_Store

Untracked files:
    Untracked:  formatted_data.txt
    Untracked:  output/viper/expr_processed/

Unstaged changes:
    Modified:   output/viper/regul_object/hnsc.RDS
    Modified:   output/viper/regul_object/kirc.RDS
    Modified:   output/viper/regul_object/luad.RDS
    Modified:   output/viper/regul_object/lusc.RDS
    Modified:   output/viper/regul_object/paad.RDS
    Modified:   output/viper/regulon/regulonblca_SYMBOL.adj
    Modified:   output/viper/regulon/regulonbrca_SYMBOL.adj
    Modified:   output/viper/regulon/reguloncesc_SYMBOL.adj
    Modified:   output/viper/regulon/reguloncoad_SYMBOL.adj
    Modified:   output/viper/regulon/regulonesca_SYMBOL.adj
    Modified:   output/viper/regulon/regulongbm_SYMBOL.adj
    Modified:   output/viper/regulon/regulonhnsc_SYMBOL.adj
    Modified:   output/viper/regulon/regulonkirc_SYMBOL.adj
    Modified:   output/viper/regulon/regulonkirp_SYMBOL.adj
    Modified:   output/viper/regulon/regulonlaml_SYMBOL.adj
    Modified:   output/viper/regulon/regulonlihc_SYMBOL.adj
    Modified:   output/viper/regulon/regulonluad_SYMBOL.adj
    Modified:   output/viper/regulon/regulonlusc_SYMBOL.adj
    Modified:   output/viper/regulon/regulonnet_SYMBOL.adj
    Modified:   output/viper/regulon/regulonov_SYMBOL.adj
    Modified:   output/viper/regulon/regulonpaad_SYMBOL.adj
    Modified:   output/viper/regulon/regulonpcpg_SYMBOL.adj
    Modified:   output/viper/regulon/regulonprad_SYMBOL.adj
    Modified:   output/viper/regulon/regulonread_SYMBOL.adj
    Modified:   output/viper/regulon/regulonsarc_SYMBOL.adj
    Modified:   output/viper/regulon/regulonstad_SYMBOL.adj
    Modified:   output/viper/regulon/regulontgct_SYMBOL.adj
    Modified:   output/viper/regulon/regulonthca_SYMBOL.adj
    Modified:   output/viper/regulon/regulonthym_SYMBOL.adj
    Modified:   output/viper/regulon/regulonucec_SYMBOL.adj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/viper_3.Rmd</code>) and HTML
(<code>docs/viper_3.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zhenzuo2/PPP/blob/fe48b9faedf94f04f17e3fef08e1689b4349bdf3/analysis/viper_3.Rmd" target="_blank">fe48b9f</a>
</td>
<td>
Zhen Zuo
</td>
<td>
2024-05-24
</td>
<td>
Publish files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level1">
<h1>Load packages</h1>
<pre class="r"><code>library(viper)</code></pre>
</div>
<div id="generating-the-gene-expression-signatures"
class="section level1">
<h1>Generating the gene expression signatures</h1>
<pre class="r"><code>dset &lt;- readRDS(&quot;output/viper/expr_processed/kirc.RDS&quot;)
signature &lt;- rowTtest(dset, &quot;description&quot;, &quot;tumor&quot;, &quot;normal&quot;)
signature &lt;- (qnorm(signature$p.value/2, lower.tail = FALSE) * sign(signature$statistic))[, 1]</code></pre>
</div>
<div id="null-model-by-sample-permutations" class="section level1">
<h1>NULL model by sample permutations</h1>
<pre class="r"><code>nullmodel &lt;- ttestNull(dset, &quot;description&quot;, &quot;tumor&quot;, &quot;normal&quot;, per = 1000, repos = TRUE, verbose = FALSE)</code></pre>
</div>
<div id="msviper" class="section level1">
<h1>msVIPER</h1>
<pre class="r"><code>regulon &lt;- readRDS(&quot;output/viper/regul_object/kirc.RDS&quot;)
regulon</code></pre>
<pre><code>Object of class regulon with 3045 regulators, 9693 targets and 107201 interactions</code></pre>
<div id="normalized-enrichment-score-nes" class="section level2">
<h2>Normalized Enrichment Score (NES)</h2>
<pre class="r"><code>mrs &lt;- msviper(signature, regulon, nullmodel, verbose = FALSE)
summary(mrs)</code></pre>
<pre><code>        Regulon Size   NES  p.value    FDR
EGF         EGF   43 -3.66 2.48e-04 0.0307
STAP1     STAP1   35 -3.67 2.39e-04 0.0307
DMRT2     DMRT2  101 -3.68 2.33e-04 0.0307
FGF9       FGF9   53 -3.69 2.26e-04 0.0307
TMPRSS2 TMPRSS2   59 -3.74 1.85e-04 0.0307
KNG1       KNG1   57 -3.82 1.33e-04 0.0307
GCGR       GCGR   65 -3.84 1.23e-04 0.0307
FOXI1     FOXI1   83 -3.92 8.99e-05 0.0307
PIK3C2G PIK3C2G   32 -3.97 7.13e-05 0.0307
OXGR1     OXGR1   89 -4.04 5.34e-05 0.0307</code></pre>
</div>
</div>
<div id="beyond-msviper" class="section level1">
<h1>Beyond msVIPER</h1>
<div id="leading-edge-analysis" class="section level2">
<h2>Leading-edge analysis</h2>
<p>msVIPER infers the relative activity of a regulatory gene based on
the enrichment of its most closely regulated targets on a given GES, but
does not identify which are the target genes enriched in the GES.
Subramanian et al. proposed a method called leading-edge analysis to
identify the genes driving the enrichment of a gene-set on a GES based
on Gene Set Enrichment Analysis (GSEA). We implemented the leading-edge
analysis in the ledge function of the viper package. The function only
has a
<code>msviper' class object as argument and generates an updated</code>msviper’
object that now includes a `ledge’ slot.</p>
<pre class="r"><code>mrs &lt;- ledge(mrs)
summary(mrs)</code></pre>
<pre><code>        Regulon Size   NES  p.value    FDR
EGF         EGF   43 -3.66 2.48e-04 0.0307
STAP1     STAP1   35 -3.67 2.39e-04 0.0307
DMRT2     DMRT2  101 -3.68 2.33e-04 0.0307
FGF9       FGF9   53 -3.69 2.26e-04 0.0307
TMPRSS2 TMPRSS2   59 -3.74 1.85e-04 0.0307
KNG1       KNG1   57 -3.82 1.33e-04 0.0307
GCGR       GCGR   65 -3.84 1.23e-04 0.0307
FOXI1     FOXI1   83 -3.92 8.99e-05 0.0307
PIK3C2G PIK3C2G   32 -3.97 7.13e-05 0.0307
OXGR1     OXGR1   89 -4.04 5.34e-05 0.0307
                                               Ledge
EGF               DCLK1, SKA3, GNG7, TTR, + 34 genes
STAP1             KCNE1, LY9, CHAC1, KIT, + 25 genes
DMRT2     PART1, TPD52L1, CKMT2, C6orf52, + 88 genes
FGF9    ADRB1, TUBB2A, PRR15L, LINC01587, + 44 genes
TMPRSS2     C1orf168, GRHL1, WSCD2, RPF2, + 48 genes
KNG1       TDRD5, SNX33, ZCCHC16, PTPN13, + 49 genes
GCGR      HELZ2, NDRG2, TMSB15B, COL28A1, + 56 genes
FOXI1    COQ3, PTPN3, APOBEC3F, C1orf168, + 61 genes
PIK3C2G     ALDH1A2, CLUL1, STAP1, CEP55, + 27 genes
OXGR1   C9orf84, LZTS1, ADRB1, LINC01587, + 77 genes</code></pre>
</div>
<div id="bootstrap-msviper" class="section level2">
<h2>Bootstrap msVIPER</h2>
<pre class="r"><code>signature &lt;- bootstrapTtest(dset, &quot;description&quot;, &quot;tumor&quot;, &quot;normal&quot;, per = 1000, verbose = FALSE)
mrs &lt;- msviper(signature, regulon, nullmodel, verbose = FALSE)</code></pre>
<pre class="r"><code>mrs &lt;- bootstrapmsviper(mrs, &quot;mode&quot;)
summary(mrs)</code></pre>
<pre><code>        Regulon Size   NES  p.value    FDR
STAP1     STAP1   35 -3.65 2.57e-04 0.0325
DMRT2     DMRT2  101 -3.66 2.57e-04 0.0325
EGF         EGF   43 -3.66 2.52e-04 0.0325
FGF9       FGF9   53 -3.67 2.40e-04 0.0325
TMPRSS2 TMPRSS2   59 -3.70 2.17e-04 0.0325
KNG1       KNG1   57 -3.81 1.40e-04 0.0325
GCGR       GCGR   65 -3.81 1.37e-04 0.0325
FOXI1     FOXI1   83 -3.89 9.89e-05 0.0325
PIK3C2G PIK3C2G   32 -3.96 7.39e-05 0.0325
OXGR1     OXGR1   89 -4.02 5.75e-05 0.0325</code></pre>
</div>
<div id="shadow-analysis" class="section level2">
<h2>Shadow analysis</h2>
<pre class="r"><code>mrshadow &lt;- shadow(mrs, regulators = 25, verbose = FALSE)</code></pre>
<pre class="r"><code>summary(mrshadow)</code></pre>
<pre><code>$msviper.results
        Regulon Size   NES  p.value    FDR
EGF         EGF   36 -3.63 2.83e-04 0.0124
NRL         NRL   63 -3.67 2.43e-04 0.0124
STAP1     STAP1   28 -3.70 2.12e-04 0.0124
ASB5       ASB5   30 -3.72 2.02e-04 0.0124
KNG1       KNG1   39 -3.72 1.97e-04 0.0124
TMPRSS2 TMPRSS2   45 -3.79 1.48e-04 0.0124
OXGR1     OXGR1   70 -3.81 1.37e-04 0.0124
GCGR       GCGR   53 -3.87 1.10e-04 0.0124
FOXI1     FOXI1   67 -3.88 1.04e-04 0.0124
PIK3C2G PIK3C2G   29 -3.98 7.02e-05 0.0124

$Shadow.pairs
[1] &quot;OXGR1 -&gt; FGF9&quot;    &quot;OXGR1 -&gt; DMRT2&quot;   &quot;OXGR1 -&gt; NR0B2&quot;   &quot;PIK3C2G -&gt; DMRT2&quot;
[5] &quot;FOXI1 -&gt; KNG1&quot;    &quot;GCGR -&gt; DMRT2&quot;   </code></pre>
</div>
<div id="synergy-analysis" class="section level2">
<h2>Synergy analysis</h2>
<pre class="r"><code>mrs &lt;- msviperCombinatorial(mrs, verbose = FALSE)
mrs &lt;- msviperSynergy(mrs, verbose = FALSE)
summary(mrs)</code></pre>
<pre><code>                      Regulon Size   NES  p.value      FDR  Synergy
VGLL1--ABCG1     VGLL1--ABCG1   30 12.71 5.42e-37 7.39e-34 4.18e-25
PDLIM1--SRGAP3 PDLIM1--SRGAP3   27 12.46 1.23e-35 8.39e-33 3.49e-07
DMRT2--BMP7       DMRT2--BMP7   30  4.57 4.94e-06 2.24e-03 7.32e-04
PAK6--VGLL1       PAK6--VGLL1   30  4.46 8.31e-06 2.83e-03 2.94e-04
BMP7--VGLL1       BMP7--VGLL1   41  4.32 1.53e-05 3.82e-03 1.87e-06
PRDM16--VGLL1   PRDM16--VGLL1   29  4.28 1.87e-05 3.82e-03 5.95e-04
GPC3--VGLL1       GPC3--VGLL1   25  4.27 1.96e-05 3.82e-03 7.52e-04
ADGRF1--SRGAP3 ADGRF1--SRGAP3   27  4.03 5.65e-05 8.37e-03 4.18e-05
TFAP2B--VGLL1   TFAP2B--VGLL1   39  4.01 6.14e-05 8.37e-03 3.22e-06
OXGR1                   OXGR1   89 -4.02 5.75e-05 8.37e-03       NA</code></pre>
</div>
</div>
<div
id="virtual-inference-of-protein-activity-by-enriched-regulon-analysis-viper"
class="section level1">
<h1>Virtual Inference of Protein-activity by Enriched Regulon analysis
(VIPER)</h1>
<pre class="r"><code>vpres &lt;- viper(dset, regulon, verbose = FALSE)</code></pre>
<pre class="r"><code>tmp &lt;- rowTtest(vpres, &quot;description&quot;, &quot;tumor&quot;, &quot;normal&quot;)
data.frame(Gene = rownames(tmp$p.value), t = round(tmp$statistic, 2), &quot;p-value&quot; = signif(tmp$p.value, 3))[order(tmp$p.value)[1:10], ]</code></pre>
<pre><code>           Gene      t   p.value
KNG1       KNG1 -60.76 1.25e-118
FOXI1     FOXI1 -59.71 2.25e-117
DMRT2     DMRT2 -59.57 3.28e-117
TMPRSS2 TMPRSS2 -58.73 3.41e-116
FGF9       FGF9 -58.31 1.12e-115
GCGR       GCGR -57.31 1.91e-114
OXGR1     OXGR1 -57.27 2.12e-114
PIK3C2G PIK3C2G -56.97 5.15e-114
EGF         EGF -56.94 5.62e-114
PAK5       PAK5 -56.59 1.53e-113</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] viper_1.38.0        Biobase_2.64.0      BiocGenerics_0.50.0
[4] workflowr_1.7.1    

loaded via a namespace (and not attached):
 [1] gtable_0.3.5       xfun_0.44          bslib_0.7.0        ggplot2_3.5.1     
 [5] htmlwidgets_1.6.4  processx_3.8.4     lattice_0.22-6     callr_3.7.6       
 [9] vctrs_0.6.5        tools_4.4.0        ps_1.7.6           generics_0.1.3    
[13] parallel_4.4.0     tibble_3.2.1       proxy_0.4-27       fansi_1.0.6       
[17] pkgconfig_2.0.3    Matrix_1.7-0       KernSmooth_2.23-24 data.table_1.15.4 
[21] lifecycle_1.0.4    compiler_4.4.0     stringr_1.5.1      git2r_0.33.0      
[25] mixtools_2.0.0     munsell_0.5.1      getPass_0.2-4      httpuv_1.6.15     
[29] htmltools_0.5.8.1  class_7.3-22       sass_0.4.9         yaml_2.3.8        
[33] lazyeval_0.2.2     plotly_4.10.4      later_1.3.2        pillar_1.9.0      
[37] jquerylib_0.1.4    whisker_0.4.1      tidyr_1.3.1        MASS_7.3-60.2     
[41] cachem_1.1.0       nlme_3.1-164       tidyselect_1.2.1   digest_0.6.35     
[45] stringi_1.8.4      kernlab_0.9-32     dplyr_1.1.4        purrr_1.0.2       
[49] splines_4.4.0      rprojroot_2.0.4    fastmap_1.2.0      grid_4.4.0        
[53] colorspace_2.1-0   cli_3.6.2          magrittr_2.0.3     survival_3.6-4    
[57] utf8_1.2.4         e1071_1.7-14       scales_1.3.0       promises_1.3.0    
[61] segmented_2.1-0    rmarkdown_2.27     httr_1.4.7         evaluate_0.23     
[65] knitr_1.46         viridisLite_0.4.2  rlang_1.1.3        Rcpp_1.0.12       
[69] glue_1.7.0         rstudioapi_0.16.0  jsonlite_1.8.8     R6_2.5.1          
[73] fs_1.6.4          </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
