---
title: "Pipeline"
output:
  workflowr::wflow_html:
    number_sections: true
    toc: true
    toc_float: true
    theme: united
    highlight: textmate
editor_options:
  chunk_output_type: console
---

```{r}
library(png)
img <- readPNG("data/Figure2.png")
grid::grid.raster(img)
```

# Pipeline for Omic Dataset Integration

## Find Differentially Phosphorylated Peptides
- Get phosphorylation data. 
- Replicates were averaged. 
- Run t-test on naive and benign samples to identify differentially phosphorylated peptide (FDA < 0.1). 
- Map differentially phosphorylated peptide to known kinases. Save it and pass to the next step.  

### Output
- Known kinases from mapped differentially phosphorylated peptide. 

## Apply the Master Regulator Inference Algorithm (MARINa)
*We first applied the master regulator inference algorithm (MARINa) (Alvarez et al., 2015), a method to infer the activity of a given protein based on the differential expression/phosphorylation of the targets it regulates.*

### Input
- Known kinases from mapped differentially phosphorylated peptide
- Gene Expression data for normal and tumor
- Gene level pathway network (Object of class regulon with XXX regulators, XXX targets and XXX interactions). The paper used Genome-wide cross-species interrogation of disease-specific regulatory networks from 10.1016/j.ccr.2014.03.017.
- Phosphorylation data for normal and tumor.
- Phosphorylation level pathway network (Object of class regulon with XXX regulators, XXX targets and XXX interactions).

### Output
- Transcription factors with differential activity (repression/activation). 
- Differentially activated kinase regulator (only higher in tumor, based on the predicted upstream kinases for each phosphopeptide). 

*This allowed us to identify transcription factors with differential activity (repression/activation) as well as differentially activated kinase regulators (based on the predicted upstream kinases for each phosphopeptide) in metastatic CRPC samples as compared with treatment naive prostate cancers (Data S1G and S1H).*. 

*In addition, kinases directly identified by the mass spectrometer in our phosphoproteomic dataset (phosphorylated kinases) were merged with the kinase regulators before input to TieDIE.*

### Mechanisms and Details
Given a data matrix and a regulon, by running multiple samples together (similar to MARINa), `mrs <- msviper(signature, regulon, nullmodel, verbose = FALSE)` to get top regulators with Normalized Enrichment Score (NES) and p-value.  
In this section, `msviper` was used to get the output.  
*In other words, it performs a KS test for each regulator using as input the differential activity (in metastatic CRPC vs treatment naive prostate cancer) of its targets.  It applies a test based on the Kolmogorov-Smirnov test to assess whether the differential activity of activated targets have higher levels, and/or the differential activity of inhibited targets have lower levels, than expected by chance based on a uniform distribution. In the case of TF regulators, the expression levels of the targets are used for this inference; in the case of kinase regulators, the phosphorylation levels are used for the target levels. To facilitate this analysis, we combined multiple databases of predicted kinase-substrate predicted interactions to produce a comprehensive ‘regulome’ of candidate regulator kinases that are predicted to phosphorylate at least 25 proteins on at least one site (Drake et al., 2012; Lachmann and Ma’ayan, 2009). We ran the MARINa algorithm (Alvarez et al., 2015), to find ‘kinase regulators’ with significantly higher activity--as inferred from the peptides they are predicted to phosphorylate--in CRPC samples compared to the control primary and benign tumor samples. For the TF targets we used a predetermined interactome of transcription factor-to-target regulatory edges, inferred using a diverse sample of normal, primary and metastatic prostate cancer samples as well as cell lines (Aytes et al., 2014a).*


## Copy Number and Mutation Analysis 
### Output
The output will a gene list including all putative cancer driver genes.
- Identify predicted driver genes with MutSig 2.0 (Lawrence et al., 2013), and filtered these genes by intersecting with the most recent COSMIC census, representing 572 cancer-related genes according to our prior knowledge.  
- Significant focal amplifications and significant focal deletion events with driver potential, identified by the GISTIC algorithm (Mermel et al., 2011).  
*When combined with the genes found in the CNV analysis of metastatic samples, this resulted in 96 candidate genes with significant copy number alteration and 112 genes with either mutation or copy number events with driver potential. Of these, 108 had at least one copy-number or mutation event in the 49 metastatic samples.*

<span style="color:red">Copy number and mutation analysis result was used for cancer-specific network but not copy number and mutation analysis.</span>.

## TieDIE Pathway Analysis of Clinical Prostate Cancer Samples 
### INPUT
- Known kinases from mapped differentially phosphorylated peptide
- Kinase Regulators. 
- Putative cancer driver genes. 
- Transcription factors. 
- pre-defined pathway interactions, such as ‘Multinet’ (Khurana et al., 2013), which can be accessed at http://homes.gersteinlab.org/Khurana-PLoSCompBio-2013/.  

### Parameters and Trimming
*A scaffold network was selected by identifying the smallest network with high precision in terms of capturing the input genes. TieDIE computes a ratio of overlap (ROO) that reflects the proximity of genes to one another in the given input set in a given network. One can control the size of TieDIE’s resulting network via the method’s alpha parameter. Thus, we chose the alpha corresponding to the smallest network in which slightly larger and smaller networks resulted in ROO values that were less distinguishable from a random background that used permuted networks. The network producing the first significant local maximum in the ROO between the observed and background was selected for use in all subsequent analysis (Fig. S2b). Our intuition was that this procedure would provide smaller, interpretable networks that maintain a high level of specificity. The final scaffold network was derived by further restricting to protein-protein edges with at least one pair of quantified phosphopeptides with at least a modest correlation (+/- 0.3, Spearman Rho), resulting in a final ‘scaffold’ network of 122 nodes and 256 edges (190 protein-protein interaction; 131 phosphorylation). The robustness of this network was quantified by removing both proteomic and geneexpression data for each patient with a complete dataset and recomputing the “scaffold” using the same procedure described above (Fig. S4l).*

## Sample-Specific Activity and Network Generation 
*For each sample, we searched all paths connecting any active kinase, mutation or high-level copy number gain or deletion to any active TF over edges contained in the scaffold network, using the NetworkX python package (Schult and Swart, 2008) and up to an edge-depth of 4;*

### Input 
- Known kinases from mapped differentially phosphorylated peptide
- Kinase Regulators. 
- Transcription factors. 
- pre-defined pathway interactions, such as ‘Multinet’ (Khurana et al., 2013), which can be accessed at http://homes.gersteinlab.org/Khurana-PLoSCompBio-2013/.  

`vpres <- viper(dset, regulon, verbose = FALSE)`, the viper function generates a matrix or "ExpressionSet" object in case an "ExpressionSet" object is given as input of regulator's activity, containing 621 regulators \times 211 samples in our example.  

### Mechanisms and Details
*We used the VIPER package (Alvarez et al., 2015) to infer sample-specific activity (pseudo z-scores) of each of each of the 14 kinase regulators that were found to have significantly higher average activity in CRPC samples, using the 11 treatment naive or benign samples to compute the reference distribution of activity scores. Similarly, we used VIPER to generate inferences for each of the 74 TFs, for each sample, using the same microarray data, regulon and reference samples used in the MARINa analysis on the Grasso dataset. Briefly, VIPER infers the activity of a regulator in a sample by inspecting the degree to which the regulator’s targets are activated or inactivated in that sample. It applies a test based on the Kolmogorov-Smirnov test to assess whether the predicted activated targets have higher levels, and/or the predicted inhibited targets have lower levels, than expected by chance based on a uniform distribution. In the case of TF regulators, the expression levels of the targets are used for this inference; in the case of kinase regulators, the phosphorylation levels are used for the targets. Importantly, we observed a strong correspondence between inferred and observed differentially phosphorylated kinases, with a greater than 3-fold improvement in the overlap compared to what was expected by chance. However, the direct overlap between TF and kinase targets was coincidental as 21 of the 74 TF regulators were found to be directly phosphorylated by one of the 14 kinase regulators, which equals chance expectation (p<0.5, permutation test). Importantly, the TF and kinase sets were found to be significantly associated in pathway space by longer paths in the network as described below.*


