---
title: "Virtual Inference of Protein-activity by Enriched Regulon analysis (Preparing Phosphopeptides Data)"
output:
  workflowr::wflow_html:
    toc: true
    toc_float: true
    theme: cosmo
    highlight: textmate
editor_options:
  chunk_output_type: console
---

# Introduction
The VIPER (Virtual Inference of Protein-activity by Enriched Regulon analysis) algorithm allows computational inference of protein activity, on an individual sample basis, from gene expression profile data. It uses the expression of genes that are most directly regulated by a given protein, such as the targets of a transcription factor (TF), as an accurate reporter of its activity.  

We have shown that analysis of TF targets inferred by the ARACNe algorithm, using the Master Regulator Inference algorithm (MARINA), is effective in identifying drivers of specific cellular phenotype which could be experimentally validated.

# Load packages
```{r message=FALSE, warning=FALSE}
library(viper)
library(aracne.networks)
library(dplyr)
library(plyr)
library(stringr)
library(Biobase)
library(EnsDb.Hsapiens.v86)
library(foreach)
library(doParallel)
dir.create("output/viper/",showWarnings = F)
dir.create("output/viper/regulon/",showWarnings = F)
dir.create("output/viper/phos/",showWarnings = F)
```

# Extract and process phosphopeptides data

##  RNA-seq and network pairs
```{r}
df <- read.csv("data/viper/omics_regulon_pairs.csv")
knitr::kable(df)
```

## Preprocess
```{r}
for (i in 1:nrow(df)) {
    exprsFile <- df$phos[i]
    exprs <- read.table(exprsFile, header = TRUE, sep = "\t",
        row.names = 1, as.is = TRUE)
    split_rownames <- strsplit(rownames(exprs), "\\|")
    split_rownames_df <- data.frame(t(data.frame(split_rownames)))
    colnames(split_rownames_df) <- c("ENSG", "ENSP", "S","Seq","Index")
    exprs <- exprs[!duplicated(split_rownames_df$ENSG),]
    rownames(exprs) <- sub("\\..*$", "", rownames(exprs))
    geneID <- ensembldb::select(EnsDb.Hsapiens.v86, keys = rownames(exprs),
        keytype = "GENEID", columns = c("SYMBOL", "ENTREZID", "GENEID"))
    rownames(exprs) <- plyr::mapvalues(rownames(exprs), from = geneID$GENEID,
        to = geneID$SYMBOL, warn_missing = FALSE)
    exprs <- exprs[!sapply(rownames(exprs), function(x) startsWith(x, "ENSG")),
        ]
    # Identify duplicated row names
    duplicated_rows <- duplicated(rownames(exprs))
    # Remove rows with duplicated row names
    cleaned_exprs <- exprs[!duplicated_rows, , drop = FALSE]
    saveRDS(cleaned_exprs, paste("output/viper/phos/", df$Lable[i], ".RDS", sep = ""))
}
```
