---
title: "Virtual Inference of Protein-activity by Enriched Regulon analysis (Preparing Gene Expression Data)"
output:
  workflowr::wflow_html:
    toc: true
    toc_float: true
    theme: cosmo
    highlight: textmate
editor_options:
  chunk_output_type: console
---

# Load packages
```{r message=FALSE, warning=FALSE}
library(viper)
library(aracne.networks)
library(dplyr)
library(plyr)
library(stringr)
library(Biobase)
library(EnsDb.Hsapiens.v86)
library(foreach)
library(doParallel)
dir.create("output/viper/exprs/",showWarnings = F)
```

# Extract and process gene expression signatures

##  RNA-seq and network pairs
```{r}
df <- read.csv("data/viper/omics_regulon_pairs.csv")
knitr::kable(df)
```

## Preprocess
```{r}
for (i in 1:nrow(df)) {
    exprsFile <- df$RNA[i]
    exprs <- as.matrix(read.table(exprsFile, header = TRUE, sep = "\t",
        row.names = 1, as.is = TRUE))
    rownames(exprs) <- sub("\\..*$", "", rownames(exprs))
    geneID <- ensembldb::select(EnsDb.Hsapiens.v86, keys = rownames(exprs),
        keytype = "GENEID", columns = c("SYMBOL", "ENTREZID", "GENEID"))
    rownames(exprs) <- plyr::mapvalues(rownames(exprs), from = geneID$GENEID,
        to = geneID$SYMBOL, warn_missing = FALSE)
    exprs <- exprs[!sapply(rownames(exprs), function(x) startsWith(x, "ENSG")),
        ]
    # Identify duplicated row names
    duplicated_rows <- duplicated(rownames(exprs))
    # Remove rows with duplicated row names
    cleaned_exprs <- exprs[!duplicated_rows, , drop = FALSE]
    saveRDS(cleaned_exprs, paste("output/viper/exprs/", df$Lable[i], ".RDS", sep = ""))
}
```
