---
title: "Virtual Inference of Protein-activity by Enriched Regulon analysis (Master Regulator Analysis)"
output:
  workflowr::wflow_html:
    toc: true
    toc_float: true
    theme: cosmo
    highlight: textmate
editor_options:
  chunk_output_type: console
---

# Load packages
```{r message=FALSE, warning=FALSE}
library(viper)
library(foreach)
library(doParallel)
```

# Run Viper on Gene Expression Data
```{r}
# Set up parallel backend
registerDoParallel(10)
# Loop through items in parallel
foreach(name = c("kirc", "hnsc", "lusc", "luad", "paad")) %dopar% {
    dset <- readRDS(paste("output/viper/expr_processed/", name, ".RDS",
        sep = ""))
    signature <- rowTtest(dset, "description", "tumor", "normal")
    signature <- (qnorm(signature$p.value/2, lower.tail = FALSE) * sign(signature$statistic))[,
        1]

    # NULL model by sample permutations

    nullmodel <- ttestNull(dset, "description", "tumor", "normal", per = 1000,
        repos = TRUE, verbose = FALSE)

    # msVIPER

    regulon <- readRDS(paste("output/viper/expr_regul_object/", name, ".RDS",
        sep = ""))
    regulon

    ## Normalized Enrichment Score (NES)
    dir.create(paste("output/TieDIE/input/exprs_", name, "/", sep = ""), showWarnings = FALSE)
    mrs <- msviper(signature, regulon, nullmodel, verbose = FALSE)
    temp <- summary(mrs, length(mrs$regulon))
    temp$Sign <- ifelse(temp$NES > 0, "+", "-")
    temp <- temp[(temp$FDR < 0.1), ]
    temp <- temp[, c("Regulon", "NES", "Sign")]
    temp$NES <- abs(temp$NES)
    
    pathways <- read.csv("data/pc-hgnc.sif", sep = "\t", header = F)
    write.table(temp, paste("output/TieDIE/input/exprs_",
        name, "/downstream.input", sep = ""), quote = F, sep = "\t", row.names = F,
        col.names = F)
    
    pathways <- pathways[(pathways$V1 %in% temp$Regulon) | (pathways$V3 %in%
        temp$Regulon), ]
    write.table(pathways, paste("output/TieDIE/input/exprs_", name, "/pathway.sif",
        sep = ""), quote = F, sep = "\t", row.names = F, col.names = F)
}
```

# Run Viper on Phosphoproteome Data
```{r}
# Set up parallel backend
registerDoParallel(10)
# Loop through items in parallel
foreach(name = c("kirc", "hnsc", "lusc", "luad", "paad")) %dopar% {
    dset <- readRDS(paste("output/viper/phos_processed/", name, ".RDS",
        sep = ""))
    signature <- rowTtest(dset, "description", "tumor", "normal")
    signature <- (qnorm(signature$p.value/2, lower.tail = FALSE) * sign(signature$statistic))[,
        1]

    # NULL model by sample permutations

    nullmodel <- ttestNull(dset, "description", "tumor", "normal", per = 1000,
        repos = TRUE, verbose = FALSE)

    # msVIPER

    regulon <- readRDS(paste("output/viper/phos_regul_object/", name, ".RDS",
        sep = ""))
    regulon

    ## Normalized Enrichment Score (NES)
    dir.create(paste("output/TieDIE/input/phos_", name, "/", sep = ""), showWarnings = FALSE)
    mrs <- msviper(signature, regulon, nullmodel, verbose = FALSE)
    temp <- summary(mrs, length(mrs$regulon))
    temp$Sign <- ifelse(temp$NES > 0, "+", "-")
    temp <- temp[(temp$FDR < 0.1), ]
    temp <- temp[, c("Regulon", "NES", "Sign")]
    temp$NES <- abs(temp$NES)

    pathways <- read.csv("data/pc-hgnc.sif", sep = "\t", header = F)
    write.table(temp, paste("output/TieDIE/input/phos_",
        name, "/downstream.input", sep = ""), quote = F, sep = "\t", row.names = F,
        col.names = F)

    pathways <- pathways[(pathways$V1 %in% temp$Regulon) | (pathways$V3 %in%
        temp$Regulon), ]
    write.table(pathways, paste("output/TieDIE/input/phos_", name, "/pathway.sif",
        sep = ""), quote = F, sep = "\t", row.names = F, col.names = F)
}
```